File: DynamicKey/AgoraDynamicKey/rust/examples/chat_token_builder.rs
use agora_token::chat_token_builder;
use std::env;

fn main() {
    // Need to set environment variable AGORA_APP_ID
    let app_id = env::var("AGORA_APP_ID").unwrap();
    // Need to set environment variable AGORA_APP_CERTIFICATE
    let app_certificate = env::var("AGORA_APP_CERTIFICATE").unwrap();

    let user_uuid = "a7180cb0-1d4a-11ed-9210-89ff47c9da5e";
    let expire = 600;

    println!("App Id: {}", app_id);
    println!("App Certificate: {}", app_certificate);
    if app_id.is_empty() || app_certificate.is_empty() {
        println!("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE");
        return;
    }

    match chat_token_builder::build_chat_app_token(&app_id, &app_certificate, expire) {
        Ok(result) => println!("ChatAppToken: {}", result),
        Err(err) => println!("{}", err),
    }

    match chat_token_builder::build_chat_user_token(&app_id, &app_certificate, user_uuid, expire) {
        Ok(result) => println!("ChatUserToken: {}", result),
        Err(err) => println!("{}", err),
    }
}

File: DynamicKey/AgoraDynamicKey/lua/examples/chat_token_builder.lua
local chat_token_builder = require("agora_token.chat_token_builder")

-- Need to set environment variable AGORA_APP_ID
local app_id = os.getenv("AGORA_APP_ID") or ""
-- Need to set environment variable AGORA_APP_CERTIFICATE
local app_certificate = os.getenv("AGORA_APP_CERTIFICATE") or ""

local user_uuid = "a7180cb0-1d4a-11ed-9210-89ff47c9da5e"
local expire = 600

print("App Id: " .. app_id)
print("App Certificate: " .. app_certificate)
if app_id == "" or app_certificate == "" then
    print("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE")
    return
end

local status, token = pcall(function()
    return chat_token_builder.build_chat_app_token(app_id, app_certificate, expire)
end)
if status then
    print("ChatAppToken: " .. token)
else
    print("Error: " .. token)
end

status, token = pcall(function()
    return chat_token_builder.build_chat_user_token(app_id, app_certificate, user_uuid, expire)
end)
if status then
    print("ChatUserToken: " .. token)
else
    print("Error: " .. token)
end
